src/app/home/components/DataList/Plans/PlanStatus.tsx

```diff
@@ -1,59 +1,47 @@
 import React from 'react';
-import { Progress, ProgressSize } from '@patternfly/react-core';
 
-const PlanStatus = ({ plan, ...props }) => {
-  const getStatus = () => {
-    const status = { text: 'Waiting for status...', progress: null };
-    let hasReadyCondition = null;
-    let hasErrorCondition = null;
-    let hasRunningMigrations = null;
-    let hasSucceededMigration = null;
-    let hasPrevMigrations = null;
-    let hasClosedCondition = null;
-    if (plan.MigPlan.status) {
-      hasClosedCondition = !!plan.MigPlan.spec.closed;
-      hasReadyCondition = !!plan.MigPlan.status.conditions.filter(c => c.type === 'Ready').length;
-      hasErrorCondition = !!plan.MigPlan.status.conditions.filter(c => c.category === 'Critical')
-        .length;
-      if (plan.Migrations.length > 0) {
-        hasPrevMigrations = !!plan.Migrations.length;
-        hasRunningMigrations = !!plan.Migrations.filter(m => {
-          if (m.status) {
-            return m.status.conditions.some(c => c.type === 'Running');
-          }
-        }).length;
-
-        hasSucceededMigration = !!plan.Migrations.filter(m => {
-          if (m.status) {
-            return m.status.conditions.some(c => c.type === 'Succeeded');
-          }
-        }).length;
-      }
+const PlanStatus = ({ plan }) => {
+  const {
+    hasClosedCondition,
+    hasReadyCondition,
+    hasNotReadyCondition,
+    hasRunningMigrations,
+    hasSucceededMigration,
+    finalMigrationComplete,
+    hasSucceededStage,
+    hasPrevMigrations,
+    hasFailedCondition,
+    latestType,
+  } = plan.PlanStatus;
 
-      if (hasErrorCondition || !hasReadyCondition) {
-        status.text = 'Not Ready';
-      }
-      if (hasReadyCondition || !hasPrevMigrations) {
-        status.text = 'Ready';
-      }
-      if (hasRunningMigrations) {
-        status.text = 'Running';
-      }
-      if (hasSucceededMigration) {
-        status.text = 'Succeeded';
-      }
-      if (hasClosedCondition) {
-        status.text = 'Closed';
-      }
+  const getStatus = () => {
+    const status = { text: 'Waiting for status...' };
+    if (hasReadyCondition || !hasPrevMigrations) {
```

Comment:
The way this ladder of if statements is structured, it's going to set the text in order of precedence from top to bottom. so if more than one applies, the lowest one is going to be the one that's set. I don't know if that's the intention or not, but just wanted to call that out. If we want to short circuit the first match, a switch with breaks might be a better choice.

Reply:
Yeah this was something I was meaning to ask you about. The logic gets a bit hairy here so I did order them in a specific way. For example, If there is a ready condition and there are migrations running, i want the running status to be shown rather than ready even though the plan is still ready.

Reply:
Cool, just wanted to make sure there wasn't unintended behavior here. At some point it'd be cool to spend some time refactoring this so it's a little more maintainable and cleaner, but I think that's out of scope for this PR. Worth an issue.

Reply:
#291 filed